### DISCUSSION TITLE
Create and Update Form in one, Problem

### DISCUSSION THREAD

**Discussion by Simonkcaon** [Budibase/budibase#7929] [Help]
**Following Problem**: I have a **Form** that creates an **Entry**, but I want this **Form** under a certain condition to act like an **Update Form**. I set a Conditions that changes the Form to Update, but to populate the Form with the Entries I need to wrap it in a Data Provider and Repeater. This works fine as long as there are already entries in the Table, but if there are no entries, the Data Provider is Empty so due to the Repeater the Form itself is hidden, thus there is no way to Create a new Entry. 

Of course i could duplicate the Form, them hide and show one of the two under certain Conditions, but it would be easier if there is some way to conditionally deactivate the Repeater so that the Form is still being shown even tho the Table is empty.

Is there any way to do this already?

**Comment by Zero2XIII** [comment]
Hi Simon,

Why do you need to use a data provider and repeater to generate the form?

The main difference between a write and update form is the passing of the entry id when writing the data to the table, maybe you can make that a parameter to include if it is supplied.

Best regards,
Philip

**Reply by Simonkcaon** [reply]
Hey Philipp, 
Thank you for your Answer,

I want the User to see the Data he already entered when he changes them, so as far as i know, to pre-fill the Form with the Entries i need to wrap the Form inside the Data Provider and Repeater. 

Simon

**Reply by Zero2XIII** [reply]
Hi Simon,

Cool, try and set the "_id" field only when updating and not when in create 'mode' and see if that works?

**Reply by Simonkcaon** [reply]
Hey Philipp,
I dont really understand what you mean, sorry. Theres no problems with the functionality, creating and updating works fine. My only problem is that as long as the table is empty the repeater doesnt show the UI elements inside it. But i need the UI elements to create the first entry..

**Reply by peterhuppertz** [reply]
I'm struggleing with the same problem:
1) If I update data I provide the _id to load the existing entry in the Data Provider and add a Repeater to show the form with the existing content. This works perfect!
2) I would love to reuse the same form with empty values to enter new data. However, since the Data Provider does not return any data the Repeater element shows the empty text and the user can't enter data.

**Comment by gmentzik** [comment]
Havent tested it, but generally you could avoid repeater and take data directly from provider rows.
So Just within provider, set manually default form field values with javascript.
Something like :
if $("OfferingProvider.Rows Length") > 0 {
 return $( Provider.rows.0.smthing );
}
Take row _id again from provider {{ Provider.rows.0._id }}. At action save row id _id exists will update row. If not will create a new one.
